pkg/runtime/helper_test.go

```diff
@@ -63,6 +63,28 @@ func TestExtractList(t *testing.T) {
 	}
 }
 
+func TestExtractGenericList(t *testing.T) {
```

Comment:
@smarterclayton I don't know what you mean with []*runtime.Object, []api.Foo is already tested in TestExtractList and []*api.Foo is what I'm testing